### PR TITLE
Validations and tests PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this branch, I;

- Added the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Added unit specs for all of my models' methods and validations.
- Fixed linter errors

Looking forward to hearing from you.

Thanks.

### CODE REVIEW

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @mertkantaroglu, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is missing a section of README, (RUN TEST). It will be good to make instructions or how to run tests for your project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  it 'name is present' do
+    expect(user).to be_valid
+    expect(user.name).to eql 'Mert'
+  end
+
+  it 'should have a link for photo' do
+    expect(user.photo).to_not eql ''
+    expect(user.photo).to eql 'www.unsplash.com'
+  end
```
Comment (codepantha) [inline spec/user_spec.rb:14]: - [x] According to requirements, every custom method in the models should have a corresponding unit test.
Having said that, please write a test for the `last_posts` method in your `User` model

File: spec/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  it 'checks the post author id' do
+    expect(post.author_id).to eql 1
+  end
+
+  it 'checks title presence for post' do
+    post.title = ''
+    expect(post).to_not be_valid
+    expect(post.title).to eql ''
+  end
```
Comment (codepantha) [inline spec/post_spec.rb:14]: - [x]  According to requirements, every custom method in the models should have a corresponding unit test.
Having said that, please write tests for the `increase_post_counter` and `last_comments` methods in your `Post` model

File: spec/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  it 'has a author id number' do
+    expect(comment.author_id).to_not eql(-3)
+    expect(comment.author_id).to eql 1
+  end
+
+  it 'has a post id number' do
+    expect(comment.post_id).to_not eql(-5)
+    expect(comment.post_id).to eql 1
+  end
```
Comment (codepantha) [inline spec/comment_spec.rb]: - [x]  According to requirements, every custom method in the models should have a corresponding unit test.
Having said that, please write a test for the `increase_comment_counter` method in your Comment model

To do this, you can assert that the `comments_counter` for a post is updated whenever the a comment is made on a post.

File: spec/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  it 'has an author id' do
+    expect(like.author_id).to_not eql 5
+    expect(like.author_id).to eql 1
+  end
+
+  it 'has an post id' do
+    expect(like.post_id).to_not eql(-5)
+    expect(like.post_id).to eql 1
+  end
+end
```
Comment (codepantha) [inline spec/like_spec.rb]: - [x]  According to requirements, every custom method in the models should have a corresponding unit test.
Having said that, please write a test for the `increase_like_counter` method in your `Like` model.

To do this, you can assert that the `likes_counter` for a post is updated whenever the post is liked.

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @mertkantaroglu 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added model validations 📌 
✔️ Added unit tests 📌 
✔️ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — COMMENTED** [review]
Hi @mertkantaroglu 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Linetrs are passing ✔️ 
👍🏽 Comments from the previous review have been addressed ✔️ 
👍🏽 PR is descriptive ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]

**Comment by mertkantaroglu** [comment]
Thank you @codepantha  and @descholar-ceo 👍

**Pull request merged**